fe/fe-core/src/main/java/com/starrocks/catalog/ColocateTableIndex.java

```diff
@@ -771,14 +771,78 @@ public long saveColocateTableIndex(DataOutputStream dos, long checksum) throws I
         return checksum;
     }
 
+    public List<GroupId> getOtherGroupsWithSameOrigName(String origName, long dbId) {
+        List<GroupId> groupIds = new ArrayList<>();
+        try {
+            readLock();
+            for (String key : groupName2Id.keySet()) {
```

Comment:
Seems groupName2Id.entrySet() is what you want? Because you call groupName2Id.get(key) multiple times.

Reply:
you're right, fixed